src/lib/components/Button/Button.jsx

```diff
@@ -39,20 +46,20 @@ export const Button = React.forwardRef((props, ref) => {
       className={classNames(
         styles.root,
         getRootPriorityClassName(
-          resolveContextOrProp(context && context.priority, priority),
+          resolveContextOrProp(buttonGroupContext && buttonGroupContext.priority, priority),
           styles,
         ),
         getRootColorClassName(color, styles),
         getRootSizeClassName(
-          resolveContextOrProp(context && context.size, size),
+          resolveContextOrProp(primaryContext && primaryContext.size, size),
           styles,
         ),
         getRootLabelVisibilityClassName(labelVisibility, styles),
-        resolveContextOrProp(context && context.block, block) && styles.isRootBlock,
-        context && styles.isRootGrouped,
+        resolveContextOrProp(buttonGroupContext && buttonGroupContext.block, block) && styles.isRootBlock,
+        primaryContext && styles.isRootGrouped,
```

Comment:
Here we need to distinguish between `ButtonGroup` context and `InputGroup` context. Current styling looks like it doesn't matter, but things would break with custom `ButtonGroup` theming, e.g. with a gap between buttons or with a different styling of the separator. So there will be two classes:

- `.isRootGrouped` becomes something like `.isRootInButtonGroup`,
- new class `.isRootInInputGroup` will be used for the `InputGroup` context.